core/src/main/java/com/airbnb/aerosolve/core/models/AdditiveModel.java

```diff
@@ -6,25 +6,55 @@
 import com.airbnb.aerosolve.core.ModelRecord;
 import com.airbnb.aerosolve.core.function.AbstractFunction;
 import com.airbnb.aerosolve.core.function.Function;
+import com.airbnb.aerosolve.core.function.FunctionUtil;
+import com.airbnb.aerosolve.core.function.MultiDimensionSpline;
 import com.airbnb.aerosolve.core.util.Util;
 import lombok.Getter;
 import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
 
 import java.io.BufferedReader;
 import java.io.BufferedWriter;
 import java.io.IOException;
 import java.util.*;
 
+import static com.airbnb.aerosolve.core.function.FunctionUtil.toFloat;
+
 // A generalized additive model with a parametric function per feature.
 // See http://en.wikipedia.org/wiki/Generalized_additive_model
+@Slf4j
 public class AdditiveModel extends AbstractModel {
+  private static final String DENSE_FAMILY = "dense";
   @Getter @Setter
   private Map<String, Map<String, Function>> weights = new HashMap<>();
 
+  private Map<String, Function> denseWeights;
+
+  private Map<String, Function> getOrCreateDenseWeights() {
+    if (denseWeights == null) {
+      denseWeights = new HashMap<>();
+      weights.put(DENSE_FAMILY, denseWeights);
```

Comment:
this is assuming there is only one feature family for the dense features, I feel this is not very generic